easybuild/easyblocks/c/cfdemcoupling.py

```diff
@@ -0,0 +1,169 @@
+        if len(lpp_dirs) == 1:
+            env.setvar('CFDEM_LPP_DIR', lpp_dirs[0])
+        else:
+            raise EasyBuildError("Failed to isolate LPP-* directory in %s", self.builddir)
+
+        # build in parallel
+        env.setvar("WM_NCOMPPROCS", str(self.cfg['parallel']))
+
+        vtk_root = get_software_root('VTK')
+        if vtk_root:
+            vtk_ver_maj_min = '.'.join(get_software_version('VTK').split('.')[:2])
+            vtk_inc = os.path.join(vtk_root, 'include', 'vtk-%s' % vtk_ver_maj_min)
+            if os.path.exists(vtk_inc):
+                env.setvar('VTK_INC_USR', '-I%s' % vtk_inc)
+            else:
+                raise EasyBuildError("Expected directory %s does not exist!", vtk_inc)
+
+            vtk_lib = os.path.join(vtk_root, 'lib')
+            if os.path.exists(vtk_lib):
+                env.setvar('VTK_LIB_USR', '-L%s' % vtk_lib)
+            else:
+                raise EasyBuildError("Expected directory %s does not exist!", vtk_lib)
+        else:
+            raise EasyBuildError("VTK not included as dependency")
+
+        # can't seem to use defined 'cfdemSysTest' alias, so call cfdemSystemTest.sh script directly...
+        cmd = "source $CFDEM_bashrc && $CFDEM_SRC_DIR/lagrangian/cfdemParticle/etc/cfdemSystemTest.sh"
+        run_cmd(cmd, log_all=True, simple=True, log_ok=True)
+
+    def build_step(self):
+        """Custom build procedure for CFDEMcoupling."""
+
+        if get_software_root('OpenFOAM'):
+            openfoam_ver = get_software_version('OpenFOAM')
+            openfoam_maj_ver = openfoam_ver.split('.')[0]
+        else:
+            raise EasyBuildError("OpenFOAM not included as dependency")
+
+        # make sure expected additionalLibs_* file is available
+        addlibs_subdir = os.path.join('src', 'lagrangian', 'cfdemParticle', 'etc', 'addLibs_universal')
+        src_addlibs = os.path.join(self.cfdem_project_dir, addlibs_subdir, 'additionalLibs_%s.x' % openfoam_maj_ver)
+        target_addlibs = os.path.join(self.cfdem_project_dir, addlibs_subdir, 'additionalLibs_%s' % openfoam_ver)
+        copy_file(src_addlibs, target_addlibs)
+
+        # can't seem to use defined 'cfdemCompCFDEMall' alias...
+        cmd = "$CFDEM_SRC_DIR/lagrangian/cfdemParticle/etc/compileCFDEMcoupling_all.sh"
+        run_cmd("source $FOAM_BASH && source $CFDEM_bashrc && %s" % cmd, log_all=True, simple=True, log_ok=True)
+
+    def install_step(self):
+        """No custom install procedure for CFDEMcoupling."""
+        pass
+
+    def sanity_check_step(self):
+        """Custom sanity check for CFDEMcoupling."""
+        comp_fam = self.toolchain.comp_family()
+        if comp_fam == toolchain.GCC:
+            wm_compiler = 'Gcc'
+        elif comp_fam == toolchain.INTELCOMP:
+            wm_compiler = 'Icc'
+        else:
+            raise EasyBuildError("Unknown compiler family, don't know how to set WM_COMPILER")
+
+        psubdir = "linux64%sDPInt32Opt" % wm_compiler
```

Comment:
It is preferable to handle debug being set and use Debug (or if it is Dbg) instead of Opt if it is.
Or is this not really built?

Reply:
See the OpenFOAM easyblock for reference

Reply:
I wouldn't switch to `Dbg` if `eb --debug` is used (which is about debug logging, not debug builds).

Reply:
I meant toolchainopt debug

Reply:
I made the necessary changes to set `$WM_COMPILE_OPTION=Debug` if the `debug` toolchain option is set, but when using that the installation fails with:

```
software/binutils/2.28-GCCcore-6.4.0/bin/ld.gold: error: cannot find -lfiniteVolume
/software/binutils/2.28-GCCcore-6.4.0/bin/ld.gold: error: cannot find -lturbulenceModels
/software/binutils/2.28-GCCcore-6.4.0/bin/ld.gold: error: cannot find -lincompressibleTurbulenceModels
/software/binutils/2.28-GCCcore-6.4.0/bin/ld.gold: error: cannot find -lfvOptions
/software/binutils/2.28-GCCcore-6.4.0/bin/ld.gold: error: cannot find -lmeshTools
/software/binutils/2.28-GCCcore-6.4.0/bin/ld.gold: error: cannot find -llagrangian
/software/binutils/2.28-GCCcore-6.4.0/bin/ld.gold: error: --add-needed is not supported but is required for libvtkCommonCore-8.1.so.1 in /software/CFDEMcoupling/3.8.0-foss-2018a/CFDEMcoupling-3.8.0/platforms/linux64GccDPInt32Debug/lib/liblmp_auto-fulldebug.so                                                                                                                                                                                           collect2: error: ld returned 1 exit status
```

Also, I don't it's worth the effort at all to try and add support for debug builds in this case...

Reply:
so lets ignore it...